### PR TITLE
Using 'nameof' operator instead of magic strings

### PR DESCRIPTION
**Pull request by hishamco**
Using the "nameof" operator instead of magic string will help us if the variable change in the future for whatever the reason

### CODE REVIEW

File: src/Microsoft.AspNet.Routing/Constraints/LengthRouteConstraint.cs [hunk 1]
```diff
@@ -24,7 +24,7 @@ public LengthRouteConstraint(int length)
             if (length < 0)
             {
                 var errorMessage = Resources.FormatArgumentMustBeGreaterThanOrEqualTo(0);
-                throw new ArgumentOutOfRangeException("length", length, errorMessage);
+                throw new ArgumentOutOfRangeException(nameof(length), errorMessage);
```
Comment (dougbu) [inline src/Microsoft.AspNet.Routing/Constraints/LengthRouteConstraint.cs]: why remove the value from the `Exception`?

File: src/Microsoft.AspNet.Routing/RouteContext.cs [hunk 2]
```diff
@@ -31,7 +31,7 @@ public RouteData RouteData
             {
                 if (value == null)
                 {
-                    throw new ArgumentNullException("value");
+                    throw new ArgumentNullException(nameof(value));
```
Comment (dougbu) [inline src/Microsoft.AspNet.Routing/RouteContext.cs]: FYI the changed `ArgumentNullException`s in this PR could all use `[NotNull]` or (for setters) `[param: NotNull]`. But there are few enough of these that there's no reason to hold up this PR.

**Comment by dougbu** [comment]
:watch: for switch back to current `ArgumentOutOfRangeException` constructors

**Comment by hishamco** [comment]
@dougbu I will look to all the PR request that I made, if [NotNull] it will fit I will close them

**Comment by hishamco** [comment]
@dougbu I made a new PR #185 that have all the fixes

**Pull request closed without merging**